maxima.yaml

```diff
@@ -0,0 +1,39 @@
+name: maxima
+sources:
+  - type: archive
+    url: https://downloads.sourceforge.net/maxima/maxima-5.42.1.tar.gz
+    sha256: 8f555aec33bc61b5a3ee0fe2e9d6c1179db67a2ff7e0eceb6bb614058eeb40cd
+  # https://bugzilla.redhat.com/show_bug.cgi?id=837142
+  # https://sourceforge.net/tracker/?func=detail&aid=3539587&group_id=4933&atid=104933
+  # https://src.fedoraproject.org/rpms/maxima/raw/66de1aea9ace767c984ad32091ee25409e2a70be/f/maxima-5.37.1-clisp-noreadline.patch
+  - type: patch
+    path: maxima-5.37.1-clisp-noreadline.patch
+  # Build the fasl while building the executable to avoid double initialization
+  # https://src.fedoraproject.org/rpms/maxima/raw/66de1aea9ace767c984ad32091ee25409e2a70be/f/maxima-5.30.0-build-fasl.patch
+  - type: patch
+    path: maxima-5.30.0-build-fasl.patch
+  # handle multiple ldflags in ecl build
+  # https://src.fedoraproject.org/rpms/maxima/raw/66de1aea9ace767c984ad32091ee25409e2a70be/f/maxima-ecl_ldflags.patch
+  - type: patch
+    path: maxima-ecl_ldflags.patch
+  - type: shell
```

Comment:
Create autogen.sh instead of running it during build.

Reply:
@barthalion 

> Create autogen.sh instead of running it during build.

What does it matter?

Reply:
I can fix it, but first I would like to know why it is so important.

Reply:
See also:
https://github.com/flathub/com.abisource.AbiWord/blob/722be32d6f6b3efb8c03c3ce29026b1ff719929e/com.abisource.AbiWord.json#L87-L92
https://github.com/flathub/com.abisource.AbiWord/blob/722be32d6f6b3efb8c03c3ce29026b1ff719929e/com.abisource.AbiWord.json#L116-L121
https://github.com/flathub/com.abisource.AbiWord/blob/722be32d6f6b3efb8c03c3ce29026b1ff719929e/com.abisource.AbiWord.json#L216-L221
https://github.com/flathub/com.abisource.AbiWord/blob/722be32d6f6b3efb8c03c3ce29026b1ff719929e/com.abisource.AbiWord.json#L270-L275
https://github.com/flathub/com.abisource.AbiWord/blob/722be32d6f6b3efb8c03c3ce29026b1ff719929e/com.abisource.AbiWord.json#L308-L313
https://github.com/flathub/com.bladecoder.adventure-editor/blob/2e9a653ed35a354c1ef647b84dfb546d693481fa/com.bladecoder.adventure-editor.json#L30-L33
https://github.com/flathub/com.google.AndroidStudio/blob/622665cdc4bcd15139433c4194fac0fae42b7a68/com.google.AndroidStudio.json#L47-L50
https://github.com/flathub/com.skype.Client/blob/3143c44ae6b71486358e677fe3b3f91f9f1eb50b/com.skype.Client.json#L61-L64
https://github.com/flathub/com.visualstudio.code/blob/17ad970b60390f5da010f6c8e5d0f27064a61c0f/com.visualstudio.code.json#L51-L54
https://github.com/flathub/io.atom.Atom/blob/ba38abc2e641f67821e83bcd55d4a2055a8d78a8/io.atom.Atom.json#L129-L132
https://github.com/flathub/org.subsurface_divelog.Subsurface/blob/30ea6400994f0e983ff9086edcd8b08db3587e0d/org.subsurface_divelog.Subsurface.yaml#L160-L162
https://github.com/flathub/org.videolan.VLC/blob/d10af9a2182e68a910aff924771f73d29e2cd0ce/org.videolan.VLC.json#L753-L756
https://github.com/flathub/shared-modules/blob/972e24443635fab6c2ce7f8f8feadd6ec445a3bb/lua5.3/lua-5.3.2.json#L28-L31

Are you going to fix it or not?

Reply:
Damn, this was actually introduced by @nedrichards, one of the flatpak maintainers.
https://github.com/flathub/com.google.AndroidStudio/commit/9720a1dcd3ccbfd0df2ffb332e883b4bf88cb1ef#diff-e466d59ba7b410fdafeffecc17678a93R49

Reply:
This, in turn, was introduced by @alexlarsson, the Flatpak creator:
https://github.com/flathub/com.skype.Client/commit/a88b8d0949bcd455024c48e2955a18bbba8c9201#diff-83817fe5ae5d45b23fd50450a88484b9R83

Reply:
Another mistake committed by @nedrichards:
https://github.com/flathub/com.visualstudio.code/commit/c63b7fa4cf396f2513e15dd3a8940aeb54363ea6#diff-db8703b20455330a2fcf99b3a3ee3415R46

Reply:
https://github.com/flathub/flathub/pull/394#issuecomment-450751472

Reply:
Another bad example, by @TingPing:
https://github.com/flathub/io.atom.Atom/commit/45376dd3f33b8cbb929ee583b06905a3a0dc4880#diff-74c5d7a1539ea5609ea30f27df21356aR86

Reply:
This one has just been introduced by **you**! What a shame!
https://github.com/flathub/shared-modules/commit/e99cf1175925a815685b7453d5092806928b1284#diff-649bad558c1e7ecb7b3d42697a939c1cR23
Why no one here cares about proper quality?!

Reply:
@barthalion 

Answer me only one question: Why should I care about "rules" that you don't even follow?

Reply:
You keep missing the problem here, which is the way you react to any feedback about your pull requests. We all are volunteers here and don't follow a strict set rules like you are probably doing in every project you are contributing to. Instead of responding "why", you explode, send dozen of links proving that you are innocent (while nobody ever suggested your changes are bad or wrong) and expect us to have fun working with you.

You should care because it's technically correct and what upstream tarball should be doing. If you had spent the time on submitting pull requests fixing various issues, even when they are effectively cosmetic, instead of these quarrels, it would be a different story.

Reply:
@barthalion 

> You keep missing the problem here, which is the way you react to any feedback about your pull requests.

This is totally not true. I have no problem with correct criticism and valid points. I question only those that don't really matter or that are actually harmful. You know, it takes some time to rebuild the whole project and I'd rather not do this if I don't see a good reason for that.

>  We all are volunteers here and don't follow a strict set rules like you are probably doing in every project you are contributing to. Instead of responding "why", you explode, send dozen of links proving that you are innocent (while nobody ever suggested your changes are bad or wrong) and expect us to have fun working with you.

> instead of these quarrels

Treating valid remarks as an act of aggression will not lead us anywhere. :)

> You should care because it's technically correct and what upstream tarball should be doing.

Do you know why I prefer `shell` instead of `script` here? Because the first one will always be executed, regardless of whether the `configure` file exists or not. Of course, I could use `rm-configure` here, but then you would probably complain that sometimes I do it unnecessarily, because this file in the particular archive doesn't exist.
It often happened that I had to change sources, e.g. because the project moved from one site to another, or the main repository was temporarily unavailable (`ragel`, `colm`). I don't want to check every time whether given archive contains the `configure` file or not. What's more, it's easy to make a mistake here, and `autoreconf` usually matters only on `aarch64`.

> If you had spent the time on submitting pull requests fixing various issues, even when they are effectively cosmetic, instead of these quarrels, it would be a different story.

Just like here?
https://github.com/flathub/org.gnome.gbrainy/pull/4
No response at all.

Flatpak lacks extensive documentation and examples. Instead, maintainers refer to the existing projects. Then it turns out that almost all of them are **incorrect** and full of **terrible mistakes**. However, maintainers clearly don't care about fixing these so-called **bugs**. Do you think that's okay?

Anyway, you are a Flathub maintainer, not me. You should take care of these projects, not me. You already know about these shortcomings, so why not fix them yourself? Don't you care about the quality of Flathub?
It would be a different story if I became an official Flathub member, but it probably will never happen because of your personal objections.
The other thing is that I'm not really convinced about the need to make all these changes...

---

The only requested change I really question here is the layout change (merging multiple modules into a single file). It is obvious to me that you don't have time to properly maintain your own packages/modules, not to mention the other manifests on Flathub. Still you're trying to force a layout that suits you, instead of that one that will be easier to maintain for me.
It's like arguing which indent is better: a double space or tab.

Reply:
> This is totally not true. I have no problem with correct criticism and valid points. I question only those that don't really matter or that are actually harmful. You know, it takes some time to rebuild the whole project and I'd rather not do this if I don't see a good reason for that.

We have the technology to do that for you.

> Do you know why I prefer shell instead of script here? Because the first one will always be executed, regardless of whether the configure file exists or not. Of course, I could use rm-configure here, but then you would probably complain that sometimes I do it unnecessarily, because this file in the particular archive doesn't exist.

And you couldn't have said that without as first response, without behaving in a childish manner and showing that you can use organization-wise search on Github to try to prove your point.

> Treating valid remarks as an act of aggression will not lead us anywhere. :)

Don't turn it around as you always do.

> Flatpak lacks extensive documentation and examples. Instead, maintainers refer to the existing projects. Then it turns out that almost all of them are incorrect and full of terrible mistakes. However, maintainers clearly don't care about fixing these so-called bugs. Do you think that's okay?

Full of things that could have been done better, would not be accepted today or end u[ overseen because some of us are just humans. But all your *exaggerations* don't really prove anything.

> Anyway, you are a Flathub maintainer, not me.

I'm reviewer, not the maintainer. I do have to read your change, and possibly revisit in the future if you become unresponsive. You are free to ignore everything I say, as you really do most of the time anyway.

Reply:
> We have the technology to do that for you.

Technology that will take me back in time? Great! I want to try it! Will it work with a stable flatpak release or should I use a bleeding-edge snapshot instead?
I'm sorry, but I couldn't resist. ;)

> some of us are just humans

What about the others? Are they bots? ;)

> You are free to ignore everything I say, as you really do most of the time anyway.

I have the impression that no matter what I do, you will find a reason to block merging my packages, as it was in the case of ReText, where even the upstream vendor was involved.
Everything could be different if you change your approach.

---

Anyway, I have already removed the installation of license files, as you requested (I just didn't push commits yet). Should I still replace `shell` with `script`, or there's no need for that anymore? I can do this, although I prefer not.

Reply:
The question is whether every review will keep looking like this or not, because that's the problem and reason for "me blocking" your packages.